lib/infra_conversion_throttler.rb

```diff
@@ -0,0 +1,46 @@
+class InfraConversionThrottler
+  DEFAULT_EMS_MAX_RUNNERS = 10
+
+  def self.start_conversions
+    pending_conversion_jobs.each do |ems, jobs|
+      running = ems.conversion_hosts.inject(0) { |sum, ch| sum + ch.active_tasks.size }
+      slots = (ems.miq_custom_get('Max Transformation Runners') || DEFAULT_EMS_MAX_RUNNERS).to_i - running
+      jobs.each do |job|
+        task = job.migration_task
+        eligible_hosts = ems.conversion_hosts.select(&:eligible?).sort_by { |ch| ch.active_tasks.size }
+        break if slots <= 0 || eligible_hosts.empty?
+        begin
+          task.conversion_host = eligible_hosts.first
+          task.save!
```

Comment:
Can be simplified to `task.update_attributes!(:conversion_host => eligible_hosts.first)`